metals/src/main/scala/scala/meta/internal/metals/UserConfiguration.scala

```diff
@@ -260,6 +261,16 @@ object UserConfiguration {
            |shown in the hover.
            |""".stripMargin,
       ),
+      UserConfigurationOption(
```

Comment:
We could also do the same trick as with type, where we have `minimal` as an option. So for `showImplicitArguments` minimal would not show evidence, however with true it would

Reply:
I'm not really a fan of this solution in `inferred types`.
I would rather have a separate option for inferred type parameters, and use `true`/`minimal` for showing type annotations in bind:
```scala
val x = List(1) match 
  case head :: tail => 1
  case _ => 0

// with 'minimal'
val x: Int = List(1) match 
  case head :: tail => 1
  case _ => 0

// with 'true' (currently also with minimal)
val x: Int = List(1) match 
  case head: Int :: tail: List[Int] => 1
  case _ => 0
```

Reply:
I agree with @jkciesluk, doing it this way makes it far more flexible. At some point we can consider better options for the user to configure those settings but that is more of a client problem.